sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -141,6 +141,11 @@ public DefaultAndroidEventProcessor(
           event.getEventId());
     }
 
+    // userId should be set even if event is Cached as the userId is static and won't change anyway.
```

Comment:
Is it static though? I mean it is if we use ANDROID_ID or the installation ID, but if it was a user_id defined by the client, it could be a different one after the restart.

Reply:
`ANDROID_ID` won't change, only if uninstall & install the App.
`installationID` is only for sessions.

if it was an user defined by the client, it won't overwrite it, not sure if I follow your comment.

Reply:
we're setting the whole user object here though after restart

Reply:
I see now that getUser actually creates a brand new users and sets default values.

Could we rename this to `getDefaultUser` or something more clear?

Reply:
yep, done